arcade/text.py

```diff
@@ -569,6 +569,50 @@ def position(self, point: Point):
             self._label.position = *point, self._label.z
 
 
+class CustomTextSprite(object):
+    def __init__(self, string, Alphabet_Textures, scale=1, 
+                 center_x=0, center_y = 0, 
+                 text_scale=1, text_margin=16, width=100, height = 40,  Background_offset_x=0, Background_offset_y=0, Background_scale=1, Background_Texture=None) -> None:
+        super().__init__()
+        self.Sprite_List = arcade.SpriteList()
+        self.Background_Sprite = arcade.Sprite(Background_Texture, center_x=center_x+width/2+Background_offset_x, center_y=center_y-height*2+Background_offset_y, scale=Background_scale)
+        
+        self.text_scale = text_scale
+        self.width = width
+        self.height = height
+        self.update_text(string, Alphabet_Textures, scale=scale, text_scale=text_scale,
+                 center_x=center_x, center_y=center_y, 
+                 text_margin=text_margin, width=width, height = height)
+        
+    def update_text(self, text, Alphabet_Textures, scale=1, text_scale=1,
+                 center_x=0, center_y = 0, 
+                 text_margin=16, width=100, height = 40):
+        self.text = text
+        self.Sprite_List.clear()
+        if not text:
+            return
+        words = text.split(' ')
+        x = 0
+        y = 0
+
+        for word in words:
+            if x > width/2+center_x:
+                y -= text_margin
+                x = -width/2
+
+
+            for string in word:
```

Comment:
`string` isn't clear here. `char` might be clearer, or `character`. 

There's also another hard problem left unaddressed here, and I don't blame you for missing it since text rendering is surprisingly complicated: your current code doesn't accurately cover multi-character [symbols such as emoji](https://unicode.org/emoji/charts/emoji-zwj-sequences.html). Pyglet already implements some support for detecting these if I remember correctly.

Reply:
What should I do? Should I make a separate dictionary for emojis and have a thing for starting them? Like emoji_Face or @@@Face or something?

Reply:
#### tl;dr Use this class in your projects as-is, but don't try to implement general text rendering

Why?
1. [Text rendering is a very hard problem](https://faultlore.com/blah/text-hates-you/) 
2. It's easier to use a TTF font maker like [BitFontMaker2](https://www.pentacom.jp/pentacom/bitfontmaker2/) as a stop-gap until the pyglet bitmap font support situation improves. 


#### I'm still not convinced



The issue you filed is very vague and I'm not sure what you're trying to achieve.

If you just want to load bitmap fonts, I'd use this class in your own projects or polish it as an add-on module outside of arcade. I'd be glad to help you with covering more edge cases via Discord. 

The article I linked in the tl;dr doesn't even mention the difficulties you can have with libraries. I've paused [my own attempt at general text loading & rendering](https://github.com/pushfoo/fontknife) because of reliability & distribution problems with dependencies. Based on that experience, my advice on implementing your own general text rendering is the same as einarf's: don't try to implement unless you absolutely have to.

If you want to expand your class to cover as much of standard glyph rendering as possible for use in your own projects, I'd start with the same route I took:

1. Add [an expanded regex module](https://pypi.org/project/regex/) to your project to allow chunking strings with multi-character glyph support
3. Use the library to map emoji to 
4. Use a recognizable filler glyph (often called the 'tofu') to represent missing glyphs in a font atlas

This will get you far enough to duplicate a large portion of pyglet's font loading plus some extra, but without its layout features.

Adding support for bitmap fonts and emoji to arcade is hard, and more complicated than just loading the fonts. We'd have to add the bitmap support in pyglet, which has a hard rule of no external python package dependencies. That means the following functionality has to be written from scratch:

1. Well-polished bitmap font loading support that returns font atlases
2. Rewrite pyglet's string parsing to add full support for multi-character glyphs

This also doesn't cover issues like support for color emoji fonts, which are a show-stopping issue for making fontknife truly useful. Per